### PR TITLE
feat: [AXIMST-24] Course unit - Sidebar buttons functional

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
**YouTrack**
https://youtrack.raccoongang.com/issue/AXIMST-24

### CODE REVIEW

File: src/course-unit/data/api.js [hunk 1]
```diff
@@ -130,6 +131,30 @@ export async function createCourseXblock({
...
+      metadata: {
+        visible_to_staff_only: isVisible,
+      },
+    } : {}),
+  };
+
+  const { data } = await getAuthenticatedHttpClient()
+    .post(getXBlockBaseApiUrl(unitId), body);
+
+  return data;
```
Comment (monteri) [inline src/course-unit/data/api.js:155]: Should it be wrapped in camel case?

File: src/course-unit/data/thunk.js [hunk 2]
```diff
@@ -34,8 +35,9 @@ import {
   updateLoadingCourseSectionVerticalDataStatus,
   updateLoadingCourseXblockStatus,
   updateCourseVerticalChildren,
-  updateCourseVerticalChildrenLoadingStatus,
+  updateCourseVerticalChildrenLoadingStatus, updateQueryPendingStatus,
```
Comment (monteri) [inline src/course-unit/data/thunk.js:38]: Import to a new line

File: src/course-unit/data/thunk.js [hunk 3]
```diff
@@ -113,6 +115,31 @@ export function editCourseItemQuery(itemId, displayName, sequenceId) {
...
 
+export function editCourseUnitVisibilityAndData(itemId, type, isVisible) {
+  return async (dispatch) => {
+    dispatch(updateSavingStatus({ status: RequestStatus.PENDING }));
+    dispatch(updateQueryPendingStatus(true));
+    const notificationMessage = getNotificationMessage(type, isVisible);
+    dispatch(showProcessingNotification(notificationMessage));
+
+    try {
+      await handleCourseUnitVisibilityAndData(itemId, type, isVisible).then(async (result) => {
```
Comment (monteri) [inline src/course-unit/data/thunk.js:126]: Is it transferred? I would prefer to use `await`. `await` is the same as `.then(`
```suggestion
      const result = await handleCourseUnitVisibilityAndData(itemId, type, isVisible);
      ...
```

File: src/course-unit/data/utils.js [hunk 4]
```diff
@@ -211,3 +214,25 @@ export function normalizeCourseSectionVerticalData(metadata) {
...
+/**
+ * Get the notification message based on the publishing type and visibility.
+ * @param {string} type - The publishing type.
+ * @param {boolean} isVisible - The visibility status.
+ * @returns {string} The corresponding notification message.
+ */
+export const getNotificationMessage = (type, isVisible) => {
+  let notificationMessage;
+
+  if (type === PUBLISH_TYPES.discardChanges) {
```
Comment (monteri) [inline src/course-unit/data/utils.js:227]: 👍

File: src/course-unit/sidebar/index.jsx [hunk 5]
```diff
@@ -1,27 +1,51 @@
...
+    dispatch(editCourseUnitVisibilityAndData(blockId, PUBLISH_TYPES.discardChanges));
+  };
+
+  const handleCourseUnitPublish = () => {
+    dispatch(editCourseUnitVisibilityAndData(blockId, PUBLISH_TYPES.makePublic));
+  };
 
   return (
     <Card
+      data-testid={isDisplayUnitLocation ? 'course-unit-location-sidebar' : 'course-unit-sidebar'}
```
Comment (monteri) [inline src/course-unit/sidebar/index.jsx:45]: We can avoid passing testing staff here by accepting `const Sidebar = ({ blockId, isDisplayUnitLocation, ...props })` and then in the test we can pass `data-testid`.

Best practice is to pass `data-testid` in the component only if there is no way to do props spreading.

File: src/course-unit/data/api.js [hunk 6]
```diff
@@ -9,6 +9,7 @@ import {
   appendBrowserTimezoneToUrl,
   normalizeCourseSectionVerticalData,
 } from './utils';
+import { PUBLISH_TYPES } from '../constants';
```
Comment (ihor-romaniuk) [inline src/course-unit/data/api.js:12]: [nit] Let's move it above of './utils' import.

File: src/course-unit/data/thunk.js (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply src/course-unit/data/thunk.js:126]: We use this approach in all the thunks on this page, let's not stand out from the crowd)

File: src/course-unit/sidebar/index.jsx (hunk 5 above)
Comment (PKulkoRaccoonGang) [reply src/course-unit/sidebar/index.jsx:45]: Agree, thanks 👍

File: src/course-unit/data/api.js (hunk 6 above)
Comment (PKulkoRaccoonGang) [reply src/course-unit/data/api.js:12]: Moved, thanks

**Review by monteri — APPROVED** [review]

**Pull request merged**